pkg/api/controller_test.go

```diff
@@ -3463,3 +3463,127 @@ func TestController_ClientDisconnect(t *testing.T) {
 		t.Fatalf("Metric for client request closed: %d, expected: %d", clientRequestClosedCount, expectedCount)
 	}
 }
+
+func TestController_SendStatsEvents(t *testing.T) {
+	clt, deps := setupClientWithAdmin(t)
+	ctx := context.Background()
+
+	tests := []struct {
+		name               string
+		events             []api.StatsEvent
+		expectedStatusCode int
+	}{
+		{
+			name: "single_event_count_1",
+			events: []api.StatsEvent{
+				{
+					Class: "single_event_count_1",
+					Name:  "name",
+					Count: 1,
+				},
+			},
+			expectedStatusCode: http.StatusNoContent,
+		},
+		{
+			name: "single_event_count_gt_1",
+			events: []api.StatsEvent{
+				{
+					Class: "single_event_count_gt_1",
+					Name:  "name",
+					Count: 3,
+				},
+			},
+			expectedStatusCode: http.StatusNoContent,
+		},
+		{
+			name: "multiple_events",
+			events: []api.StatsEvent{
+				{
+					Class: "class_multiple_events_ev_1",
+					Name:  "name1",
+					Count: 1,
+				},
+				{
+					Class: "class_multiple_events_ev_2",
+					Name:  "name2",
+					Count: 1,
+				},
+			},
+			expectedStatusCode: http.StatusNoContent,
+		},
+		{
+			name: "empty_usage_class",
+			events: []api.StatsEvent{
+				{
+					Class: "",
+					Name:  "name",
+					Count: 1,
+				},
+			},
+			expectedStatusCode: http.StatusBadRequest,
+		},
+		{
+			name: "empty_usage_name",
+			events: []api.StatsEvent{
+				{
+					Class: "class_empty_usage_name",
+					Name:  "",
+					Count: 1,
+				},
+			},
+			expectedStatusCode: http.StatusBadRequest,
+		},
+		{
+			name: "zero_usage_count",
+			events: []api.StatsEvent{
+				{
+					Class: "class_zero_usage_count",
+					Name:  "name",
+					Count: 0,
+				},
+			},
+			expectedStatusCode: http.StatusNoContent,
+		},
+		{
+			name: "negative_usage_count",
+			events: []api.StatsEvent{
+				{
+					Class: "class_negative_usage_count",
+					Name:  "name",
+					Count: -23,
+				},
+			},
+			expectedStatusCode: http.StatusBadRequest,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			resp, err := clt.SendStatsEventsWithResponse(ctx, api.SendStatsEventsJSONRequestBody{
+				Events: tt.events,
+			})
+			if err != nil {
+				t.Fatalf("SendStatsEvents failed: %s", err)
+			}
+
+			if resp.StatusCode() != tt.expectedStatusCode {
+				t.Fatalf("SendStatsEvents status code: %d, expected: %d", resp.StatusCode(), tt.expectedStatusCode)
+			}
+
+			// For successful requests, verify the expected events were sent
+			if tt.expectedStatusCode == http.StatusNoContent {
```

Comment:
I think we can remove the check to validate that we didn't find any event that match the class/name.
It will require `expectedCount` that we will match the "count" of event we found.

Reply:
done